examples/src/main/scala/com/cloudera/livy/examples/WordCountApp.scala

```
@@ -0,0 +1,243 @@
+   * arg(3) - Socket stream port that spark streaming will listen to.
+   * Optional arguments need to be passed in a key=value way
+   * Example:
+   * host=examplehost where "host" is the required key for arg(2)
+   * port=8080 where "port" is the required key for arg(3)
+   *
+   * STEPS FOR EXECUTION - To get accurate results for one execution:
+   * 1) Delete if the file already exists in the given outputFilePath.
+   *
+   * 2) Spark streaming will listen to the given or defaulted host and port. So textdata needs to be
+   *    passed as socket stream during the run-time of the App. The streaming context reads 2
+   *    batches of data with an interval of 20 seconds for each batch. All the data has to be
+   *    fed before the streaming context completes the second batch. NOTE - Inorder to get accurate
+   *    results for one execution, pass the textdata before the execution of the app so that all the
+   *    data is read by the socket stream.
+   *    To pass data to localhost and port 8086 provide the following command
+   *    nc -kl 8086
+   *
+   * 3) The text can be provided as paragraphs as the app will tokenize the data and filter spaces.
+   *
+   * 4) Execute the application jar file with the required and optional arguments either using
+   * mvn or scala.
+   *
+   * Example execution:
+   * scala -cp /pathTo/livy-api-*version*-SNAPSHOT.jar:/pathTo/livy-client-http-*version*
+   * -SNAPSHOT.jar:/pathTo/livy-examples-*version*-SNAPSHOT.jar:/pathTo/livy-scala-api-*version*
+   * -SNAPSHOT.jar com.cloudera.livy.examples.WordCountApp http://livy-host:8998 /outputFilePath
+   * host=myhost port=8080
+   */
+  def main(args: Array[String]) {
+
+    var socketStreamHost: String = "localhost"
+    var socketStreamPort: Int = 8086
+    var url = ""
+    var outputFilePath = ""
+    def parseOptionalArg(arg: String): Unit = {
+      val Array(argKey, argValue) = arg.split("=")
+      if (argKey == "host") {
```

you can use `argKey match { case "host" => ...` instead, and you should also error out if an unknown argument is provided.